src/toniarts/openkeeper/world/room/Temple.java

```diff
@@ -102,19 +102,18 @@ protected BatchNode constructFloor() {
             int i = count / waterArea[0].length;
             int j = count % waterArea[0].length;
 
-//            if(waterArea[i][j]) {
-//                Terrain terrainAtPoint = roomInstance.getTerrainAtPoint(p);
-//                terrainAtPoint.getFlags().add(Terrain.TerrainFlag.WATER);
-//                EntityInstance<Terrain> entityInstance = new EntityInstance<>(terrainAtPoint);
-//
-//                waterBatches.add(entityInstance);
-//            }
+            if(waterArea[i][j]) {
+                EntityInstance<Terrain> ent = new EntityInstance<>(getWorldState().getGameState().getLevelData().getMap().getWater());
```

Comment:
We have had unbatched version of the map rendering at the beginning, my GeForce 970 was brought to its knees when we had all the objects and the map. We didn't even have creatures back then. The ye olden days.